### PR TITLE
Add new metrics for NEG DualStack Migrator

### PR DESCRIPTION
**Pull request by gauravkghildiyal**
/assign @swetharepakula

### CODE REVIEW

**Comment by gauravkghildiyal** [comment]
Ready

File: pkg/neg/metrics/neg_metrics_collector.go [hunk 1]
```diff
@@ -205,3 +241,124 @@ func (sm *SyncerMetrics) computeEndpointStateMetrics(forDegradedMode bool) (negt
...
+		dualStackEndpointType: dualStackCount,
+		migrationEndpointType: migrationCount,
+	}
+}
+
+func (sm *SyncerMetrics) computeDualStackMigrationDurations() ([]int, int) {
+	sm.mu.Lock()
+	defer sm.mu.Unlock()
+
+	finishedDurations, longestUnfinishedDuration := make([]int, 0), 0
```
Comment (swetharepakula) [inline pkg/neg/metrics/neg_metrics_collector.go:321]: Can you add a comment, either here or in the metric definition that if this is 0, that means no migrations are currently ongoing?

File: pkg/neg/metrics/neg_metrics_collector.go [hunk 2]
```diff
@@ -104,6 +120,21 @@ func (sm *SyncerMetrics) export() {
 	}
 
 	sm.logger.V(3).Info("Exporting syncer related metrics", "Syncer count", syncerCount, "Number of Endpoints", lpMetrics.NumberOfEndpoints)
+
+	finishedDurations, longestUnfinishedDurations := sm.computeDualStackMigrationDurations()
+	for _, duration := range finishedDurations {
+		DualStackMigrationFinishedDurations.Observe(float64(duration))
```
[the remark this answers is not in the document]
Comment (gauravkghildiyal) [reply pkg/neg/metrics/neg_metrics_collector.go:126]: Great catch!
Yes I deleted it when we emit. Thanks for noticing.
Added coverage in test.

File: pkg/neg/metrics/neg_metrics_collector.go (hunk 1 above)
Comment (gauravkghildiyal) [reply pkg/neg/metrics/neg_metrics_collector.go:321]: Sure thing. Added in the metrics description.

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**